rhel/vex/fetcher.go

```diff
@@ -0,0 +1,384 @@
+				return nil, hint, fmt.Errorf("error writing compacted JSON to tmp file: %w", err)
+			}
+			entriesWritten++
+		}
+		if !errors.Is(err, io.EOF) {
+			return nil, hint, fmt.Errorf("error reading tar contents: %w", err)
+		}
+
+		zlog.Debug(ctx).
+			Str("updater", u.Name()).
+			Int("entries written", entriesWritten).
+			Msg("finished writing compressed data to spool")
+	}
+
+	err = u.processChanges(ctx, cw, fp)
+	if err != nil {
+		return nil, hint, err
+	}
+
+	err = u.processDeletions(ctx, cw, fp)
+	if err != nil {
+		return nil, hint, err
+	}
+
+	fp.version = updaterVersion
+	fp.requestTime = time.Now()
+	success = true
+	return f, driver.Fingerprint(fp.String()), nil
+}
+
+// ProcessChanges deals with the published changes.csv, adding records
+// to w means they are deemed to have changed since the compressed
+// file was last processed. w and fp can be modified.
+func (u *Updater) processChanges(ctx context.Context, w io.Writer, fp *fingerprint) error {
+	uri, err := u.url.Parse(changesFile)
+	if err != nil {
+		return err
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, uri.String(), nil)
+	if err != nil {
+		return err
+	}
+	if fp.changesEtag != "" {
+		req.Header.Add("If-None-Match", fp.changesEtag)
+	}
+	res, err := u.client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	switch res.StatusCode {
+	case http.StatusOK:
+		if t := fp.changesEtag; t == "" || t != res.Header.Get("etag") {
+			break
+		}
+		fallthrough
+	case http.StatusNotModified:
+		return nil
+	default:
+		return fmt.Errorf("unexpected response from changes.csv: %s", res.Status)
+	}
+	fp.changesEtag = res.Header.Get("etag")
+
+	rd := csv.NewReader(res.Body)
+	rd.FieldsPerRecord = 2
+	rd.ReuseRecord = true
+	var (
+		l       int
+		buf, bc bytes.Buffer
+	)
+	rec, err := rd.Read()
+	for ; err == nil; rec, err = rd.Read() {
+		buf.Reset()
+		bc.Reset()
+		if len(rec) != 2 {
+			return fmt.Errorf("could not parse changes.csv file")
+		}
+
+		cvePath, uTime := rec[0], rec[1]
+		year, err := strconv.ParseInt(path.Dir(cvePath), 10, 64)
+		if err != nil {
+			return fmt.Errorf("error parsing year %w", err)
+		}
+		if year < lookBackToYear {
+			continue
+		}
+		updatedTime, err := time.Parse(time.RFC3339, uTime)
+		if err != nil {
+			return fmt.Errorf("line %d: %w", l, err)
+		}
+		if updatedTime.Before(fp.requestTime) {
+			continue
+		}
+
+		advisoryURI, err := u.url.Parse(cvePath)
+		if err != nil {
+			return err
+		}
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, advisoryURI.String(), nil)
+		if err != nil {
+			return fmt.Errorf("error creating advisory request %w", err)
+		}
+
+		// Use a func here as we're in a loop and want to make sure the
+		// body is closed in all events.
+		err = func() error {
+			res, err := u.client.Do(req)
+			if err != nil {
+				return fmt.Errorf("error making advisory request %w", err)
+			}
+			defer res.Body.Close()
+			err = httputil.CheckResponse(res, http.StatusOK)
```

Comment:
just curious why this is used here but not earlier in the file where similar status checks were used

Reply:
Added for the latest request and the request of the compressed data, didn't add it for the others (changes.csv and deletions.csv) as status informs the logic going forward.